### PR TITLE
Generate single html

### PR DESCRIPTION
**Pull request by EncodePanda**
After this change we get:

1. better UX by providing arguments with flags
2. single html generated from multiple sources
3. templating mechanism (user can provide its own styles)
4. table of contents automatically generated (if `$toc$` present somewhere in the template)

Example usage:

* asking for help
```
$> frost --help                                                                                                         
Frost - automatically generates documentation from your source code

Usage: frost [--input STRING]... [--template STRING] --output STRING

Available options:
  -h,--help                Show this help text
  --input STRING...        Files from which documentation is generated. Accepts
                           multiple values
  --template STRING        (optional) HTML template used to generate the
                           documentation
  --output STRING          Path to a file that will hold generated documentation
```

* creating documenation with template (that has table of content in it)

```
$> frost --input ./examples/git.md  \ 
               --input ./examples/python.md \
               --output doc.html 
               --template ./examples/standalone.html
```

gives us

image

### CODE REVIEW

**Comment by lukasz-golebiewski** [comment]
Cool stuff! Travis is unhappy though

**Comment by EncodePanda** [comment]
@lukasz-golebiewski Travis is now happy :)

File: examples/template.html [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                    </div>
+                </div>
+
+                <div class="uk-width-medium-3-4">
+$body$                    
+                </div>
+            </div>
+$if(analytics)$                    
+            <script>
+                  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
```
Comment (lukasz-golebiewski) [inline examples/template.html]: What is google analytics needed for?

File: app/Main.hs [hunk 2]
```diff
@@ -48,9 +49,9 @@ main = do
   where
     exit ExitSuccess     = exitSuccess
     exit (ExitFailure 1) = exitFailure
-    generate (Config filePaths outputFilePath) = generateDocs (transform plugins)
+    generate (Config filePaths templatePath outputFilePath) = generateDocs (transform plugins)
```
[the remark this answers is not in the document]
Comment (lukasz-golebiewski) [reply app/Main.hs]: It's unclear to me as well

File: src/Frost/PandocRun.hs [hunk 3]
```diff
@@ -30,7 +30,7 @@ runOutputPandoc :: (
 runOutputPandoc inputFilePath = interpret $ \case
   Output pandoc -> do
     content <- fromPandocIO $ writeHtml4String def pandoc
-    writeFile (inputFilePath ++ ".html") content
+    writeFile inputFilePath content
```
[the remark this answers is not in the document]
Comment (lukasz-golebiewski) [reply src/Frost/PandocRun.hs]: It's unclear to me as well

[the remark this answers is not in the document]
Comment (EncodePanda) [reply src/Frost/PandocRun.hs]: yeah, renamed

[the remark this answers is not in the document]
Comment (EncodePanda) [reply src/Frost/PandocRun.hs]: this btw old code :)

File: examples/template.html (hunk 1 above)
Comment (EncodePanda) [reply examples/template.html]: it's a template from the webs, we can remove that

File: app/Main.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (EncodePanda) [reply app/Main.hs]: fixed

File: examples/template.html [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        <link rel="apple-touch-icon-precomposed" href="images/apple-touch-icon.png">
+
+$if(template_css)$
+<link rel="stylesheet" href="$template_css$">
+$else$
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/uikit/2.26.4/css/uikit.gradient.css">
+$endif$
+
+        <link rel="stylesheet" href="style.css">
+        <link href="https://vjs.zencdn.net/5.4.4/video-js.css" rel="stylesheet" />
```
Comment (lukasz-golebiewski) [inline examples/template.html]: Do you intend to display videos here? :-D
https://videojs.com/getting-started/

**Comment by EncodePanda** [comment]
Template from https://github.com/diversen/pandoc-uikit

**Pull request merged**